docs/src/main/asciidoc/cdi.adoc

```diff
@@ -0,0 +1,361 @@
+////
+This guide is maintained in the main Quarkus repository
+and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/master/docs/src/main/asciidoc
+////
+= Quarkus - Introduction to Contexts and Dependency Injection
+
+include::./attributes.adoc[]
+:numbered:
+:sectnums:
+:sectnumlevels: 4
+:toc:
+
+In this guide we're going to describe the basic principles of the programming model used all over the Quarkus universe. 
+Quarkus programming model is based on the http://docs.jboss.org/cdi/spec/2.0/cdi-spec.html[Contexts and Dependency Injection for Java 2.0, window="_blank"] specification.
+However, a full CDI implementation is not used under the hood.
+Quarkus only implements a subset of the CDI features - see also <<cdi-reference.adoc#supported_features,the list of supported features>> and <<cdi-reference.adoc#limitations,the list of limitations>>.
+On the other hand, there are quite a few <<cdi-reference#nonstandard_features,non-standard features>> and <<cdi-reference.adoc#build_time_apis,Quarkus-specific APIs>>.
+We believe that our efforts will drive the innovation of the CDI specification towards the build-time oriented developer stacks in the future.
```

Comment:
I think this part is too detailed for the introduction of this guide. I would put it at the end. Keep in mind that you target people not knowing CDI so they do not really care about this.

Reply:
OK, so move all sentences after `However,...`?

Reply:
I think you need to choose between introduction to pure CDI where you would avoid Quarkus-specific examples (omitting no-args constructor for instance) and then you can move this text towards the end of the guide. Or you should state ahead that Quarkus does it slightly differently in the name of innovation and then we can keep the guide as-is. Just my 0.02$ :)

Reply:
> I think you need to choose between introduction to pure CDI...

I tend to disagree. IMHO it's ok to mix the contents. It's a quarkus guide anyway. I moved the sentences to the end.